src/ReactHardwareComponent.js

```diff
@@ -42,16 +64,23 @@ type ViewConfigPropTypes = {
  * @constructor ReactHardwareComponent
  * @extends ReactComponent
  * @extends ReactMultiChild
- * @param {!object} viewConfig View Configuration.
  */
-const ReactHardwareComponent = function(
-  element: React$Element
-) {
-  this.viewConfig = DEFAULT_VIEW_CONFIG;
+const ReactHardwareComponent = function(element: React$Element) {
   this._rootNodeID = null;
   this._renderedChildren = null;
   this._io = null;
-  this.construct(element);
+  this._currentElement = element;
+  this.viewConfig = DEFAULT_VIEW_CONFIG;
+  if (typeof element.type === 'string') {
+    this.viewConfig = getViewConfig(element.type);
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    invariant(
```

Comment:
This is what RN does: https://github.com/facebook/react/blob/master/src/renderers/native/ReactNative/ReactNativeDefaultInjection.js#L101-L108

A warning sounds good. I am not sure that defaulting to a pin is a good idea. I'd rather just warn and try to render (nothing will happen because we only call `setPayload` when it is a pin). This way people will notice that they are using an invalid type.